pkg/agent/supportbundlecollection/support_bundle_controller.go

```diff
@@ -301,100 +295,29 @@ func (c *SupportBundleController) uploadSupportBundle(supportBundle *cpv1b2.Supp
 	if err != nil {
 		return fmt.Errorf("failed to upload support bundle while getting uploader: %v", err)
 	}
-	if _, err := outputFile.Seek(0, 0); err != nil {
-		return fmt.Errorf("failed to upload support bundle to file server while setting offset: %v", err)
-	}
-	// fileServer.URL should be like: 10.92.23.154:22/path or sftp://10.92.23.154:22/path
-	parsedURL, err := parseUploadUrl(supportBundle.FileServer.URL)
-	if err != nil {
-		return fmt.Errorf("failed to upload support bundle while parsing upload URL: %v", err)
-	}
-	triesLeft := uploadToFileServerTries
-	var uploadErr error
-	for triesLeft > 0 {
-		if uploadErr = c.uploadToFileServer(uploader, supportBundle.Name, parsedURL, &supportBundle.Authentication, outputFile); uploadErr == nil {
-			return nil
-		}
-		triesLeft--
-		if triesLeft == 0 {
-			return fmt.Errorf("failed to upload support bundle after %d attempts", uploadToFileServerTries)
-		}
-		klog.InfoS("Failed to upload support bundle", "UploadError", uploadErr, "TriesLeft", triesLeft)
-		time.Sleep(uploadToFileServerRetryDelay)
-	}
-	return nil
-}
 
-func parseUploadUrl(uploadUrl string) (*url.URL, error) {
-	parsedURL, err := url.Parse(uploadUrl)
-	if err != nil {
-		parsedURL, err = url.Parse("sftp://" + uploadUrl)
-		if err != nil {
-			return nil, err
-		}
-	}
-	if parsedURL.Scheme != "sftp" {
-		return nil, fmt.Errorf("not sftp protocol")
+	if _, err := outputFile.Seek(0, 0); err != nil {
```

Comment:
I am not sure what this is for. Do you think it would make sense to have it in the `Upload` function?

Reply:
It was , before. But that would require the args in the interface changed from `io.Reader` to `afero.File`. I'm not so sure about this change so make it `io.Reader` as the one supportbundle used in the first place, seems more generic.

Reply:
ok, I agree it makes more sense to keep it here even though it should be the responsibility of the caller of `uploadSupportBundle`